lib/modules.nix

```diff
@@ -1185,6 +1214,12 @@ let
       inherit contents;
     };
 
+  /**
+    Return a definition with file location information.
```

Comment:
```suggestion
    Return a definition property that adds file location information.
    
    The file location information affects error messages and introspection data in `options`.
    By constructing `mkDefinition` properties, you can preserve this information when generically replicating definitions from one place to another.
    It overrides the default behavior of assigning the module location to all option definitions contained within it.
    
    # Inputs
    
    - `file`: ...
    
    - `value`: a definition. This may contain further properties such as `mkMerge`, etc.
    
    # Output
    
    A module system property that preserves the location of the definition.
```

Reply:
I'd not resolve this within this PR: https://github.com/NixOS/nixpkgs/pull/390983
Was already merged, maybe i was missing the rebase. 
I'll try to make a seperate PR out of this to improve mkDefinition docs